File: HamamatsuCamera/dcamsdk4/samples/cs/csAcq4/FormInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using Hamamatsu.DCAM4;

namespace csAcq4
{
    public partial class FormInfo : Form
    {
        // ----------------

        internal class datainfo
        {
            string _infokind;
            string _infovalue;

            public string Kind
            {
                get
                {
                    return _infokind;
                }
            }
            public string Value
            {
                get
                {
                    return _infovalue;
                }
            }

            public datainfo(string kind, string value)
            {
                _infokind = kind;
                _infovalue = value;
            }
        }

        // ----------------

        public FormInfo()
        {
            InitializeComponent();
        }

        void add_info(ref List<datainfo> listinfo, ref MyDcam mydcam, DCAMIDSTR idstr, string strkind)
        {
            string strvalue = mydcam.dev_getstring(idstr);
            if (strvalue.Length > 0)
            {
                listinfo.Add(new datainfo(strkind, strvalue));
            }
        }

        public void set_mydcam(ref MyDcam mydcam)
        {
            // update DataSource of DataGridView
            List<datainfo> listinfo = new List<datainfo>();

            if (mydcam != null)
            {
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.BUS, "BUS");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.CAMERAID, "CAMERAID");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.VENDOR, "VENDOR");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.MODEL, "MODEL");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.CAMERAVERSION, "CAMERA VERSION");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.DRIVERVERSION, "DRIVER VERSION");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.MODULEVERSION, "MODULE VERSION");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.DCAMAPIVERSION, "DCAM-API VERSION");
                add_info(ref listinfo, ref mydcam, DCAMIDSTR.CAMERA_SERIESNAME, "SERIES NAME");
            }
            DataGridViewInfo.DataSource = listinfo;
        }
    }
}

File: HamamatsuCamera/dcamsdk4/samples/vbnet/vbAcq4/FormInfo.vb
Imports vbAcq4.Hamamatsu.DCAM4

Public Class FormInfo

    Class datainfo
        Private _infokind As String
        Private _infovalue As String

        Public ReadOnly Property name() As String
            Get
                Return _infokind
            End Get
        End Property
        Public ReadOnly Property value() As String
            Get
                Return _infovalue
            End Get
        End Property
        Public Sub New(ByVal kind As String, ByVal value As String)
            _infokind = kind
            _infovalue = value
        End Sub
    End Class

    ' ----------------

    Sub add_info(ByRef listinfo As List(Of datainfo), ByRef mydcam As MyDcam, ByVal idstr As DCAM_IDSTR, ByVal strkind As String)
        Dim strvalue As String
        strvalue = mydcam.dev_getstring(idstr)
        If (strvalue.Length > 0) Then
            listinfo.Add(New datainfo(strkind, strvalue))
        End If
    End Sub
    Public Sub set_mydcam(ByRef mydcam As MyDcam)
        Dim listinfo As New List(Of datainfo)

        If Not mydcam Is Nothing Then
            add_info(listinfo, mydcam, DCAM_IDSTR.BUS, "BUS")
            add_info(listinfo, mydcam, DCAM_IDSTR.CAMERAID, "CAMERAID")
            add_info(listinfo, mydcam, DCAM_IDSTR.VENDOR, "VENDOR")
            add_info(listinfo, mydcam, DCAM_IDSTR.MODEL, "MODEL")
            add_info(listinfo, mydcam, DCAM_IDSTR.CAMERAVERSION, "CAMERA VERSION")
            add_info(listinfo, mydcam, DCAM_IDSTR.DRIVERVERSION, "DRIVER VERSION")
            add_info(listinfo, mydcam, DCAM_IDSTR.MODULEVERSION, "MODULE VERSION")
            add_info(listinfo, mydcam, DCAM_IDSTR.DCAMAPIVERSION, "DCAM-API VERSION")
            add_info(listinfo, mydcam, DCAM_IDSTR.CAMERA_SERIESNAME, "SERIES NAME")
        End If

        DataGridViewInfo.DataSource = listinfo
    End Sub
End Class